tests/model_serving/model_server/storage/pvc/utils.py

```diff
@@ -10,12 +12,32 @@ def create_isvc(
     name: str,
     namespace: str,
     deployment_mode: str,
-    storage_uri: str,
     model_format: str,
     runtime: str,
+    storage_uri: Optional[str] = None,
+    storage_key: Optional[str] = None,
+    storage_path: Optional[str] = None,
     min_replicas: int = 1,
     wait: bool = True,
-) -> InferenceService:
+) -> Generator[InferenceService, Any, Any]:
+    predictor_dict = {
+        "minReplicas": min_replicas,
+        "model": {
+            "modelFormat": {"name": model_format},
+            "version": "1",
+            "runtime": runtime,
+        },
+    }
+    # "type:ignore" is needed because otherwise mypy will complain about "Unsupported target for indexed assignment ("object")"
+    if storage_uri and storage_key and storage_path:
+        raise InvalidStorageArgument(storage_uri, storage_key, storage_path)
+    elif storage_uri:
+        predictor_dict["model"]["storageUri"] = storage_uri  # type: ignore
+    elif storage_key and storage_path:
+        predictor_dict["model"]["storage"] = {"key": storage_key, "path": storage_path}  # type: ignore
+    else:
```

Comment:
how about creating a helper function to check the combinations so all logic is next to each other (i.e `if storage_uri and storage_key and storage_path` and making sure that storage_key and storage_path must be provided together?